frontend/app-development/layout/PageHeader/SubHeader/SettingsModalButton/SettingsModal/components/Tabs/Maskinporten/ScopeList/ScopeList.tsx

```diff
@@ -23,3 +39,80 @@ export const ScopeList = (): ReactElement => {
     </StudioAlert>
   );
 };
+
+type ScopeListContentProps = {
+  allScopes: MaskinportenScope[];
+  selectedScopes: MaskinportenScope[];
+};
+
+const ScopeListContent = ({ allScopes, selectedScopes }: ScopeListContentProps): ReactElement => {
+  const { t } = useTranslation();
+
+  const checkboxTableRowElements: StudioCheckboxTableRowElement[] = mapScopesToRowElements(
+    allScopes,
+    selectedScopes,
+  );
+
+  // This useState is temporary to simulate correct behaviour in browser. It will be removed and replaced by a mutation function
+  const [rowElements, setRowElements] =
+    useState<StudioCheckboxTableRowElement[]>(checkboxTableRowElements);
+
+  const areAllChecked = rowElements.every((element) => element.checked);
+  const isAnyChecked = rowElements.some((element) => element.checked);
+
+  const handleChangeAllScopes = () => {
+    const updatedRowElements: StudioCheckboxTableRowElement[] = rowElements.map((element) => ({
+      ...element,
+      checked: !areAllChecked,
+    }));
+    saveUpdatedScopes(updatedRowElements);
+  };
+
+  const handleChangeScope = (event: ChangeEvent<HTMLInputElement>) => {
+    const clickedValue = event.target.value;
```

Comment:
Would it be correct to rename the variable to selectedScope? Sine the value is the scope. :)
```suggestion
    const selectedScope = event.target.value;
```